packages/relay/src/lib/eth.ts

```diff
@@ -1583,7 +1583,7 @@ export class EthImpl implements Eth {
       return null;
     }
   
-    if (blockNumberOrTag != null && blockNumberOrTag.length < 32 && !EthImpl.blockTagIsEarliestOrPending(blockNumberOrTag)) {
+    if (blockNumberOrTag != null && blockNumberOrTag.length < 32 && !EthImpl.blockTagIsEarliestOrPending(blockNumberOrTag) && !EthImpl.blockTagIsLatestOrPending(blockNumberOrTag)) {
```

Comment:
The reason this check is made is to ensure the `blockNumberOrTag` is a numerical value.
The better approach would be to do the required specific check and not call the 2 overlapping methods `EthImpl.blockTagIsEarliestOrPending` and `EthImpl.blockTagIsLatestOrPending`

Reply:
Maybe the below
```suggestion
    const blockNumber = Number(blockNumberOrTag);
    if (blockNumberOrTag != null && blockNumberOrTag.length < 32 && !isNaN(blockNumber)) {
```
Then use `blockNumber` below everywhere `Number(blockNumberOrTag)` was used

Reply:
Done.